session_0/item.rb

```diff
@@ -41,5 +49,18 @@ def discounted?
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
+    result = {
+      1 => 0,
+      2 => 0,
+      3 => 0, 
+      4 => 0,
+    }
+    items.each do |item|
+      if result[item.category_id]==0
```

Comment:
This can be simplified as just `result[item.category_id] += (item.current_price * item.quantity)` as `0 + x` equals just `x`